### PR TITLE
Replaced codespell github action as riot command, added as commit hook

### PR DESCRIPTION
**Pull request by Yun-Kim**
## Description
From #2538, `codespell` was introduced as a file spell checker for the tracer library. Instead of using `codespell` as a github action, we've added it into our `riotfile.py` as 2 riot commands, using `codespell==2.1.0` which is the current latest version of `codespell`:
- `riot run codespell` will run `codespell` across all files in `ddtrace/` and `tests/`, while skipping over `ddtrace/vendor`.
- `riot run hook-codespell [file1 file2 file3 ...]` will run `codespell` specifically on provided files. This was added as a pre-commit hook tool.

### CODE REVIEW

File: hooks/pre-commit/04-run-codespell [hunk 1]
```diff
@@ -0,0 +1 @@
+/Users/yun.kim/dd/dd-trace-py/hooks/scripts/run-codespell.sh
```
Comment (jd) [inline hooks/pre-commit/04-run-codespell:1]: That does not look  good

**Review by jd — CHANGES_REQUESTED** [review]

File: hooks/pre-commit/04-run-codespell (hunk 1 above)
Comment (brettlangdon) [reply hooks/pre-commit/04-run-codespell:1]: > works on my machine

:trollface:

Comment (Yun-Kim) [reply hooks/pre-commit/04-run-codespell]: 😭

File: riotfile.py [hunk 2]
```diff
@@ -131,6 +131,21 @@ def select_pys(min_version=MIN_PYTHON_VERSION, max_version=MAX_PYTHON_VERSION):
...
+            pkgs={"codespell": "==2.1.0"},
+            venvs=[
+                Venv(
+                    name="codespell",
+                    command="codespell --ignore-words-list='dne,fo,medias,ment,nin,ot,setttings,statics' "
+                    "--skip='ddtrace/vendor' ddtrace/ tests/",
+                ),
+                Venv(
+                    name="hook-codespell",
+                    command="codespell {cmdargs}",
```
Comment (brettlangdon) [inline riotfile.py:145]: we still need the `ignore-words-list` and `skip` here?

Comment (brettlangdon) [reply riotfile.py:145]: looks like we can setup a config file as well for the ignore words, skips, etc

https://github.com/codespell-project/codespell/#using-a-config-file

**Review by brettlangdon — APPROVED** [review]

**Review by brettlangdon — CHANGES_REQUESTED** [review]
we need to update `circleci/config.yml`

probably add this into `pre-check`?

**Review by brettlangdon — APPROVED** [review]

**Review by P403n1x87 — APPROVED** [review]

File: setup.cfg [hunk 3]
```diff
@@ -1,2 +1,6 @@
 [bdist_wheel]
 universal=1
+
+[codespell]
+skip = *.c,.riot,.tox,.mypy_cache,ddtrace/vendor
+ignore-words-list = dne,fo,medias,ment,nin,ot,setttings,statics
```
Comment (Kyle-Verhoog) [inline setup.cfg:6]: Do we still need all these ignores if we ignore the vendor directory?

`setttings` or `nin` seems like ones that shouldn't be ignored

Comment (Yun-Kim) [reply setup.cfg:6]: Screen Shot 2021-06-15 at 12 38 50 PM

Seems like these are all valid used words in our testing and integrations 😢

Comment (brettlangdon) [reply setup.cfg:6]: can we either do a PR to fix those usages first, or I am ok to merge this PR as-is, and then update to remove the skips immediately after.

**Pull request merged**